### ISSUE TITLE
EPIC: create patch-to-point workflow

### ISSUE THREAD

**Issue by kvantricht** [WorldCereal/worldcereal-classification#101]
Starting from patch extractions, we need a workflow that performs:

- [x] loading extractions as openEO datacubes
- [x] perform preprocessing
- [x] perform pixel sampling
- [x] export result as geoparquet
- [x] Optimize performance: https://github.com/Open-EO/openeo-geopyspark-driver/issues/1146

**Comment by VincentVerelst** [comment]
Currently it's not possible on the CDSE backend to `load_stac` multiple CRS: https://github.com/Open-EO/openeo-geopyspark-driver/issues/827

**Comment by VincentVerelst** [comment]
Implemented a STAC splitter in gfmap to circumvent the load_stac issue: https://github.com/Open-EO/openeo-gfmap/pull/146

**Comment by kvantricht** [comment]
Needs #125 first.

**Comment by jdries** [comment]
moving to public mount is possible, but this would trigger a huge update in stac metadata

**Comment by VincentVerelst** [comment]
- Successfully loaded precomposited METEO from STAC on Terrascope; this gives a huge performance boost
- Was able to generate a parquet file with preprocessed data of meteo, dem, s1 and s2 timeseries, using some random points
- `/vitodata/worldcereal_data` should be available on openeo-dev.vito.be soon (waiting on integration tests). This would mean we can load METEO and the Sentinel 1 and 2 patches from STAC from there
- For interaction with the RDM API we are still dependent on two issues, see: https://github.com/WorldCereal/worldcereal-classification/issues/125

**Comment by kvantricht** [comment]
That's great progress Vincent, thanks! So once the RDM interaction is there we have in principle everything in place to start gradually building this geoparquet and performing some downstream tests on them?

**Comment by VincentVerelst** [comment]
> That's great progress Vincent, thanks! So once the RDM interaction is there we have in principle everything in place to start gradually building this geoparquet and performing some downstream tests on them?

Indeed!

**Comment by kvantricht** [comment]
Conceptual pipeline:

- per ref_id per EPSG zone, get `temporal_extent` from STAC metadata filtered on the ref_id specific patches
- RDM interaction module gets geometries from RDM based on BBOX of all patches in the job and the temporal extent
- [ ] To be implemented: some way to distinguish original seed features vs collateral for downstream tracking (@cbutsko ) 
- Resulting geodataframe is written to geoparquet, containing all required attributes, to temporary folder on artifactory
- This geodataframe immediately is used (`load_url`) to launch patch-to-point script with same AOI and temporal extent
- Final result is downloaded as separate geoparquet
- [ ] To be implemented: some `post_job_action` that allows to merge this result safely into the global geoparquet

**Comment by VincentVerelst** [comment]
Blocked by gdalerrors when using `load_stac` to load in patch extractions

**Comment by HansVRP** [comment]
@kvantricht looking at the checkboxes this looks to be done and completed. What additional work do we propose here?

**Comment by HansVRP** [comment]
Closing the ticket for now as we ticked all the boxes, I propose to follow-up in a new ticket
